Makefile

```diff
@@ -1,29 +1,40 @@
 .PHONY: all build release
 
 IMAGE := goodeggs/ranch-baseimage-nodejs
-TAG_SUFFIX := -yarn
 VERSION := $(shell cat VERSION)
 MAJOR_VERSION := $(shell awk -F. '{print $$1}' VERSION)
 MINOR_VERSION := $(shell awk -F. '{print $$2}' VERSION)
 
 all: build
 
-build:
-	docker build -t $(IMAGE):latest$(TAG_SUFFIX) .
+build: build-yarn-base build-yarn build-yarn-java
+
+build-yarn-base:
+	docker build -t $(IMAGE):yarn-base .
+
+build-yarn: build-yarn-base
+	echo "FROM $(IMAGE):yarn-base" > Dockerfile.yarn
+	cat Dockerfile.onbuild.tpl >> Dockerfile.yarn
+	docker build -t $(IMAGE):yarn -f Dockerfile.yarn .
+
+build-yarn-java: build-yarn-base
+	echo "FROM $(IMAGE):yarn-base" > Dockerfile.yarn-java
+	cat Dockerfile.yarn-java.tpl >> Dockerfile.yarn-java
+	cat Dockerfile.onbuild.tpl >> Dockerfile.yarn-java
+	docker build -t $(IMAGE):yarn-java -f Dockerfile.yarn-java .
 
 release: build
-	echo $(VERSION)
-	echo $(MAJOR_VERSION)
-	echo $(MINOR_VERSION)
 	( git diff --quiet && git diff --cached --quiet ) || ( echo "checkout must be clean"; false )
-	docker run -ti -v /var/run/docker.sock:/var/run/docker.sock goodeggs/docker-squash $(IMAGE):latest$(TAG_SUFFIX)
-	docker push $(IMAGE):latest$(TAG_SUFFIX)
-	docker tag $(IMAGE):latest$(TAG_SUFFIX) $(IMAGE):$(VERSION)$(TAG_SUFFIX)
-	docker push $(IMAGE):$(VERSION)$(TAG_SUFFIX)
-	docker tag $(IMAGE):latest$(TAG_SUFFIX) $(IMAGE):$(MAJOR_VERSION).$(MINOR_VERSION)$(TAG_SUFFIX)
-	docker push $(IMAGE):$(MAJOR_VERSION).$(MINOR_VERSION)$(TAG_SUFFIX)
-	docker tag $(IMAGE):latest$(TAG_SUFFIX) $(IMAGE):$(MAJOR_VERSION)$(TAG_SUFFIX)
-	docker push $(IMAGE):$(MAJOR_VERSION)$(TAG_SUFFIX)
+	for tag in yarn yarn-java; do \
+		docker run -ti -v /var/run/docker.sock:/var/run/docker.sock goodeggs/docker-squash $(IMAGE):$$tag ; \
+		docker push $(IMAGE):$$tag ; \
+		docker tag $(IMAGE):$$tag $(IMAGE):$(VERSION)-$$tag_suffix ; \
```

Comment:
So where is `$$tag_suffix coming` from ?

Reply:
doh, fixed.